packages/react-core/package.json

```diff
@@ -34,7 +34,13 @@
   "peerDependencies": {
     "aws-amplify": "3.x.x || 4.x.x",
     "react": ">= 16.8.0",
-    "react-dom": ">= 16.8.0"
+    "react-dom": ">= 16.8.0",
+    "xstate": "^4.33.6"
+  },
+  "peerDependenciesMeta": {
+    "xstate": {
+      "optional": true
+    }
```

Comment:
`xstate` also needs to be deduped between React and react-core libraries.

Reply:
Can we just put `xstate` and `xstate/react` in _ui-react-core_ as direct deps and leave them out of _ui-react_?

Reply:
Yeah, you're right. Since react-core is handling all the xstate functionality, we can remove them from the react library.

Reply:
I need to test it since react and react-core both use the `@aws-amplify/ui` package.

Reply:
It's safe to remove xstate from the react package, done!